### ISSUE TITLE
Inherited members showing up inappropriately

### ISSUE THREAD

**Issue by analog-cbarber** [mkdocstrings/python#102]
Using mkdocstrings-python 1.5.2 w/ griffe 0.35, we noticed that suddenly inherited members
started showing up in our API docs although we did not ask for them. They show up even
when explicitly turned off.

Our code is internal, so I cannot share it, but I did some poking into `do_filter_objects`
and found that the offending members are aliases whose target is inherited. This hack
in `do_filter_objects` appears to mimic the old behavior:

```python
    if members_list is True:
        # Return all pre-selected members.
        return objects

    # HACK - remove inherited aliases
    from griffe.dataclasses import Alias
    for i in reversed(range(len(objects))):
        obj = objects[i]
        if isinstance(obj,Alias):
            assert not obj.inherited
            if obj.target.inherited:
                objects.pop(i)
```

Hope that is enough to go on....

**Comment by analog-cbarber** [comment]
I find that still leaves some broken cross references for some class methods that come from another library.

If I go into griffe and modify `Alias.members` to set the inherited property from the target:

```python
    @cached_property
    def members(self) -> dict[str, Object | Alias]:  # noqa: D102
        final_target = self.final_target
        return {name: Alias(name, target=member, parent=self, inherited=final_target.inherited) for name, member in final_target.members.items()}
```

That takes care of the remaining problem (as far as I can tell).

I assume there is a good reason that Alias does not just delegate the `inherited` property, but the issues all do seem to have to do with the inherited property of aliases....

**Comment by pawamoy** [comment]
> I assume there is a good reason that Alias does not just delegate the inherited property

There is not! Perfect catch! That's my fault, I didn't think of reapplying the property. Is this fix enough or the hack in `do_filter_objects` still necessary?

**Comment by pawamoy** [comment]
It's more like this but you did find the right place:

```python
@cached_property
def members(self) -> dict[str, Object | Alias]:  # noqa: D102
    final_target = self.final_target
    return {name: Alias(name, target=member, parent=self, inherited=False) for name, member in final_target.members.items()}


@cached_property
def inherited_members(self) -> dict[str, Alias]:  # noqa: D102
    final_target = self.final_target
    return {
        name: Alias(name, target=member, parent=self, inherited=True) for name, member in final_target.inherited_members.items()
    }
```

**Comment by pawamoy** [comment]
Fix released in v0.35.1.

**Comment by analog-cbarber** [comment]
This fix seems to do the trick.
